packages/server/migrations/20230219162719_migrate_varchar_columns_to_text.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+    await knex.raw(`
```

Comment:
I would wonder about using knex built-ins like `select` / `where` and [alterTable](https://knexjs.org/guide/schema-builder.html#altertable) for this instead? 

There's a tradeoff, in that we'd get further away from the underlying SQL, but we'd also move away from potentially error-prone raw strings, so it could go either way I guess

Reply:
eg 
```js
const tables = await knex('information_schema.tables')
    .select('table_name')
    .where('table_type', 'BASE TABLE')
    .andWhereNot('table_schema', 'information_schema')
    .andWhereNot('table_schema', 'pg_catalog')
    .andWhereNot('table_name', 'migrations');
```

Reply:
I generally do prefer to use query-builders for the reasons you mentioned, but occasionally I think the query complexity goes beyond what the builder tool does a good job supporting. 

In this case, the inner-most `SELECT` query could theoretically be produced by Knex, requiring some usage of `knex.raw` to produce the `string_agg()` call. However, I don't think that the outer `DO ... BEGIN ... EXECUTE ... END` can be produced natively by Knex, so it would end up requiring (at least one) more `knex.raw()` usages to embed the dynamic subquery. Given that using `knex.raw()` runs counter to the main benefit you mentioned in avoiding "potentially error-prone raw strings", as well as the fact that the resulting contents of the migration file would be (IMO) increasingly non-obvious, _and_ the relative level of effort to produce the inner dynamic query in the first place, I think there are diminishing returns to using the query builder here.

That said, if either of you wants to take a crack at implementing this migration in a more "Knex-native" form, I would be happy to consider it.

Reply:
I'm good with it the way it is.

Reply:
Yep this makes sense to me, thank you for reasoning through this! I will consider this as an option in future migrations also!